### ISSUE TITLE
TLSv1.2 support for MacOS desktop client

### ISSUE THREAD

**Issue by ckujau** [owncloud/client#3059]
When I try to connect OC client (1.6.4, build 1195) to a TLSv1.2-only server, the client is unable to connect: 

```Failed to connect to ownCloud at https://example.org/owncloud/status.php: Connection closed```

I'm using the SSL configuration from [Mozilla](https://mozilla.github.io/server-side-tls/ssl-config-generator/ "Mozilla SSL Configuration Generator") for Apache 2.2 and OpenSSL 1.0.1e when the error above happens.

Only when I remove the ```-TLSv1``` argument, OC client is able to connect.

Also, when I [remove the DHE ciphers](http://vincent.bernat.im/en/blog/2011-ssl-perfect-forward-secrecy.html "SSL/TLS & Perfect Forward Secrecy") from the configuration, the OC client fails to connect with:

```Failed to connect to ownCloud at https://example.org/owncloud/status.php: SSL handshake failed```
![oc_desktop_fail](https://cloud.githubusercontent.com/assets/822519/6956348/9649b078-d8a1-11e4-94de-062b5f98886b.png)

In summary, the OC client should support the following (Apache) configuration:

```
SSLProtocol all -SSLv2 -SSLv3 -TLSv1 -TLSv1.1
SSLCipherSuite "ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:DHE-DSS-AES128-GCM-SHA256:kEDH+AESGCM:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:DHE-DSS-AES128-SHA256:DHE-DSS-AES256-SHA:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!3DES:!MD5:!PSK:!DHE-RSA-AES256-GCM-SHA384:!DHE-RSA-AES128-GCM-SHA256"
```

**Comment by guruz** [comment]
> When I try to connect OC client (1.6.4, build 1195)

Please use a current version (>= 1.8)

**Comment by danimo** [comment]
1.6.4? *yuck*. That's old and was still using Apples' obsolete libssl. Please use 1.8.0, which actually uses against OpenSSL 1.0.x

**Comment by ckujau** [comment]
Strange: I downloaded the OC sync client yesterday, started it - and thought 1.6.4 was the latest version. Somehow an older versions must have snuck in. Yes, v1.8.0 supports TLSv1.2, great! :-)

Sorry for the noise.
